### ISSUE TITLE
upgrade from v2.4.6 to v.2.5.1 broken? upgraded MAZI not broadcasting SSID

### ISSUE THREAD

**Issue by mgaved** [mazi-project/portal#111]
Is there a broken upgrade path from v2.4.6 to v2.5.1?

I've installed MAZI v2.4.6 from fresh (downloaded v2.4.6 from http://nitlab.inf.uth.gr/mazi-img/)

I carry out initial installation processes (giving the MAZIzone a password, date/time, SSID...). 

The MAZIzone restarts with the new SSID, and I can login both as an end user and as an administrator. I change the network settings to dual mode and plug in an ethernet cable to my ADSL router.

I click on "Check for updates" - and it finds updates. It should be updating to v2.5.1 . IT then goes offline - as I would expect - and I wait for it to restart and broadcast its SSID.

It does not re-broadcast its SSID. I wait for approximately ten minutes then restart the MAZI manually (removing power and putting in back on again). 

It restarts but does not broadcast an SSID so I cannot connect to it.

Does anybody else have this problem? Is there a broken upgrade path from v2.4.6 to v2.5.1?

Can you advise what I should do, or further information that I can provide that would be helpful.

Thanks!

**Comment by panosnethood** [comment]
Hi Mark, I think you should wait as long as it is needed for the SSID to appear. Removing power is for sure the wrong thing to do since the update has not finished if the SSID is not broadcasted.

To be sure, just leave the default SSID during the initial setup.

**Comment by mgaved** [comment]
Thanks for your advice Panos. 
Some additional information: I have had a monitor attached to the raspberry pi and I only start to think there is an issue after it has got to the command line "login" point.  I am reporting that the start up process on the command line completes and the Raspberry Pi is waiting for username/login to be typed in at the command line, not before.

I have been leaving the upgraded Raspberry Pi for approximately ten minutes. I will try again this morning with a fresh SD card (sealed, new from shops) and wait for 30 minutes. 

I will be very happy if it's just something silly I am doing (very believable :-) ) and there's just an extra instruction I need to add to my workflow...

**Comment by haniavis** [comment]
Hi,
the time stamp of the .img file (30/05/2018 17:54) is when the file was created locally in our dev environment. The timestamp you see in the repository is when the .zip file was uploaded on the server. So it makes totally sense to be different.

This specific update process takes 2-3 minutes to complete, so as soon as you press the update button, you should wait for 3-4 minutes to let the raspberry i) reboot, ii) update dependencies iii) broadcast its WiFi. After 30min there is a timeout, and even if the update process is not finished, the WiFi SSID will appear and you will be able to connect.

Could you change the wifi SSID in the v2.4.6 before you start the update process to v2.5.1.?

**Comment by mgaved** [comment]
Thanks for the additional Info!
>Could you change the wifi SSID in the v2.4.6 before you start the update process to v2.5?

The first time I did the initial installation of v2.4.6, (including changing SSID) and then logged in , and tried the upgrade to v2.5.  This time, I have done the initial installation of v2.4.6., and pressed Submit for the initial installation without changing the SSID, and on re-connection, I have changed the SSID from the default within the admin panel. It disconnected me (as it should) and I reconnected to the new SSID, I am now in there (as admin in v2.4.6) and I'll now Check for Updates and aim to update (to v2.5.1 I assume)

**Comment by haniavis** [comment]
if you don't see the WiFi SSID there is no other solution to have access.

Since v2.5.2, the administrator is able to choose between two update channels, the stable one and the development one. 

Did you manage to get access to your mazizone?

**Comment by mgaved** [comment]
thanks for the information - the WiFi SSID did not come back. I'll try to find time to work out the issue but unlikely to report soon.

I'll download  the new v2.5.2 and start again with that. 

For deployments running in community settings it will be really important to have a stable upgrade path as users will want a safe route to updating MAZI which have gathered data collected on them (stories, photos, images etc). Thanks for indicating there will be an update channel to the stable new releases.

**Comment by mgaved** [comment]
Are there reserved characters I can't use in an SSID name? I was trying different characters and I am starting to think the issue might have been I used 'reserved characters' that I am not allowed to use and this is the reason for the WiFI SSID not broadcasting.
Underscore and hyphen seem ok
Is "." ok? how about "/"?
Is there a link to where I can see what characters are ok, and which are problematic. I will try some tests. I assume A-Z, a-z, 0-9 are all ok, and it might be other characters that are problematic. 
Perhaps if this is the issue we need a sentence describing suitable naming conventions for the MAZIzone's WiFi SSID: we need to remember we're encouraging community activists who normally might not engage with this sort of technology set-up process. 
Thanks!
Mark

**Comment by panosnethood** [comment]
Yes, ideally we should check the input string and do not allow at all invalid SSIDs and URLs as suggested in issue #53

**Comment by mgaved** [comment]
Agreed with Panos: when somebody enters an invalid WiFi SSID in the administration set up panel, there should be a validation script that responds with a message "that SSID is not allowed because you have some characters that can't be used".

The form should also indicate what characters are allowed, either text on the screen (e.g. "please only use A-Z, a-z, 0-9 characters") or as a pop up with text activated by clicking on an Information icon.

Rationale: people may not know that there are invalid SSID character strings. People may not know which characters are invalid.

See also issue #53
